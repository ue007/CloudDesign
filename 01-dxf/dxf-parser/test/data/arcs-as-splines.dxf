  0
SECTION
  2
ENTITIES
  0
SPLINE
  5
6E
330
1F
100
AcDbEntity
100
AcDbSpline
210
0.0
220
0.0
230
1.0
 70
    15
 71
     2
 72
    12
 73
     9
 74
     0
 42
0.000000001
 43
0.0000000001
 40
0.0
 40
0.0
 40
0.0
 40
17.27875959474386
 40
17.27875959474386
 40
34.55751918948772
 40
34.55751918948772
 40
51.83627878423158
 40
51.83627878423158
 40
69.11503837897544
 40
69.11503837897544
 40
69.11503837897544
 10
233.6663114494559
 20
180.0700474977344
 30
0.0
 41
1.0
 10
233.6663114494559
 20
212.3902435891571
 30
0.0
 41
0.7071067811865476
 10
201.3461153580332
 20
212.390243589157
 30
0.0
 41
1.0
 10
169.0259192666106
 20
212.3902435891571
 30
0.0
 41
0.7071067811865476
 10
169.0259192666106
 20
180.0700474977344
 30
0.0
 41
1.0
 10
169.0259192666106
 20
147.7498514063117
 30
0.0
 41
0.7071067811865476
 10
201.3461153580332
 20
147.7498514063117
 30
0.0
 41
1.0
 10
233.6663114494559
 20
147.7498514063117
 30
0.0
 41
0.7071067811865476
 10
233.6663114494559
 20
180.0700474977344
 30
0.0
 41
1.0
  0
SPLINE
  5
6F
330
1F
100
AcDbEntity
100
AcDbSpline
210
0.0
220
0.0
230
1.0
 70
    15
 71
     2
 72
    12
 73
     9
 74
     0
 42
0.000000001
 43
0.0000000001
 40
0.0
 40
0.0
 40
0.0
 40
3.926990816987241
 40
3.926990816987241
 40
7.853981633974483
 40
7.853981633974483
 40
11.78097245096172
 40
11.78097245096172
 40
15.70796326794897
 40
15.70796326794897
 40
15.70796326794897
 10
208.6916144697202
 20
243.2413398582423
 30
0.0
 41
1.0
 10
208.6916144697202
 20
250.5868389699293
 30
0.0
 41
0.7071067811865476
 10
201.3461153580332
 20
250.5868389699293
 30
0.0
 41
1.0
 10
194.0006162463463
 20
250.5868389699293
 30
0.0
 41
0.7071067811865476
 10
194.0006162463463
 20
243.2413398582423
 30
0.0
 41
1.0
 10
194.0006162463463
 20
235.8958407465554
 30
0.0
 41
0.7071067811865476
 10
201.3461153580332
 20
235.8958407465553
 30
0.0
 41
1.0
 10
208.6916144697202
 20
235.8958407465554
 30
0.0
 41
0.7071067811865476
 10
208.6916144697202
 20
243.2413398582423
 30
0.0
 41
1.0
  0
SPLINE
  5
73
330
1F
100
AcDbEntity
100
AcDbSpline
210
0.0
220
0.0
230
1.0
 70
    12
 71
     2
 72
     6
 73
     3
 74
     0
 42
0.000000001
 43
0.0000000001
 40
0.0
 40
0.0
 40
0.0
 40
17.83485687076171
 40
17.83485687076171
 40
17.83485687076171
 10
280.7529837541637
 20
303.6705820088653
 30
0.0
 41
1.0
 10
258.47226626445
 20
317.9847757821756
 30
0.0
 41
0.9841380061121457
 10
232.5957563534457
 20
323.6179587245282
 30
0.0
 41
1.0
  0
ENDSEC
  0
EOF
