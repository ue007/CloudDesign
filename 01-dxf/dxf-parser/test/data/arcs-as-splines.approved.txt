{
  "entities": [
    {
      "type": "SPLINE",
      "handle": "6E",
      "ownerHandle": "1F",
      "normalVector": {
        "x": 0,
        "y": 0,
        "z": 1
      },
      "closed": true,
      "periodic": true,
      "rational": true,
      "planar": true,
      "degreeOfSplineCurve": 2,
      "numberOfKnots": 12,
      "numberOfControlPoints": 9,
      "numberOfFitPoints": 0,
      "knotValues": [
        0,
        0,
        0,
        17.27875959474386,
        17.27875959474386,
        34.55751918948772,
        34.55751918948772,
        51.83627878423158,
        51.83627878423158,
        69.11503837897544,
        69.11503837897544,
        69.11503837897544
      ],
      "controlPoints": [
        {
          "x": 233.6663114494559,
          "y": 180.0700474977344,
          "z": 0
        },
        {
          "x": 233.6663114494559,
          "y": 212.3902435891571,
          "z": 0
        },
        {
          "x": 201.3461153580332,
          "y": 212.390243589157,
          "z": 0
        },
        {
          "x": 169.0259192666106,
          "y": 212.3902435891571,
          "z": 0
        },
        {
          "x": 169.0259192666106,
          "y": 180.0700474977344,
          "z": 0
        },
        {
          "x": 169.0259192666106,
          "y": 147.7498514063117,
          "z": 0
        },
        {
          "x": 201.3461153580332,
          "y": 147.7498514063117,
          "z": 0
        },
        {
          "x": 233.6663114494559,
          "y": 147.7498514063117,
          "z": 0
        },
        {
          "x": 233.6663114494559,
          "y": 180.0700474977344,
          "z": 0
        }
      ]
    },
    {
      "type": "SPLINE",
      "handle": "6F",
      "ownerHandle": "1F",
      "normalVector": {
        "x": 0,
        "y": 0,
        "z": 1
      },
      "closed": true,
      "periodic": true,
      "rational": true,
      "planar": true,
      "degreeOfSplineCurve": 2,
      "numberOfKnots": 12,
      "numberOfControlPoints": 9,
      "numberOfFitPoints": 0,
      "knotValues": [
        0,
        0,
        0,
        3.926990816987241,
        3.926990816987241,
        7.853981633974483,
        7.853981633974483,
        11.78097245096172,
        11.78097245096172,
        15.70796326794897,
        15.70796326794897,
        15.70796326794897
      ],
      "controlPoints": [
        {
          "x": 208.6916144697202,
          "y": 243.2413398582423,
          "z": 0
        },
        {
          "x": 208.6916144697202,
          "y": 250.5868389699293,
          "z": 0
        },
        {
          "x": 201.3461153580332,
          "y": 250.5868389699293,
          "z": 0
        },
        {
          "x": 194.0006162463463,
          "y": 250.5868389699293,
          "z": 0
        },
        {
          "x": 194.0006162463463,
          "y": 243.2413398582423,
          "z": 0
        },
        {
          "x": 194.0006162463463,
          "y": 235.8958407465554,
          "z": 0
        },
        {
          "x": 201.3461153580332,
          "y": 235.8958407465553,
          "z": 0
        },
        {
          "x": 208.6916144697202,
          "y": 235.8958407465554,
          "z": 0
        },
        {
          "x": 208.6916144697202,
          "y": 243.2413398582423,
          "z": 0
        }
      ]
    },
    {
      "type": "SPLINE",
      "handle": "73",
      "ownerHandle": "1F",
      "normalVector": {
        "x": 0,
        "y": 0,
        "z": 1
      },
      "rational": true,
      "planar": true,
      "degreeOfSplineCurve": 2,
      "numberOfKnots": 6,
      "numberOfControlPoints": 3,
      "numberOfFitPoints": 0,
      "knotValues": [
        0,
        0,
        0,
        17.83485687076171,
        17.83485687076171,
        17.83485687076171
      ],
      "controlPoints": [
        {
          "x": 280.7529837541637,
          "y": 303.6705820088653,
          "z": 0
        },
        {
          "x": 258.47226626445,
          "y": 317.9847757821756,
          "z": 0
        },
        {
          "x": 232.5957563534457,
          "y": 323.6179587245282,
          "z": 0
        }
      ]
    }
  ]
}
